src/chain/chain.rs

```diff
@@ -165,9 +165,12 @@ impl Chain {
         participants: &BTreeSet<PublicId>,
         dkg_result: &DkgResultWrapper,
     ) -> Result<(), RoutingError> {
-        if participants.contains(self.our_id()) {
-            self.new_section_bls_keys = Some(dkg_result.0.clone());
+        if let Some(first) = participants.iter().next() {
+            let _ = self
+                .new_section_bls_keys
+                .insert(*first.name(), dkg_result.0.clone());
         }
+
```

Comment:
Then it is a change, because the set have to be disjoint, since we are doing disjoint section.
Having the full set cause a bit of problem because we have to build it from the ElderInfo that do not have a set when we get it out.

Seemed to be the most straight forward way to achieve this with what we had. (Ideally we would have wanted the prefix, but not sure trying to add it would make things better overall, also bigger).

Reply:
Ah, I see - so this is only important on splits, and the two sections won't ever have the same first Elder after a split, is that the idea?
Might be worth describing this in more detail in a comment.

Reply:
Ok, is it more sensible here, or where the member field is declared?

Reply:
Hmm... I think I would add it where the member field is declared, but let's see what Adam thinks.

Reply:
The comment? Yeah, where it is declared makes more sense to me.

Reply:
Could we use the same trick (the first name as key) in `Elder`'s `dkg_cache` ?

Reply:
could probably, not as useful though since we need to create the participants set anyway, so I'd probably leave it for another time, if we decide it really make sense

Reply:
It probably makes sense only as an optimization, so not important right now.